src/components/common/header/Header.spec.js

```diff
@@ -1,13 +1,28 @@
+/* eslint-disable max-len */
 import React from 'react';
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
 import { BrowserRouter as Router } from 'react-router-dom';
-import { HeaderComponent } from './Header';
+import ConnectedHeader, { HeaderComponent } from './Header';
 import AuthenticatedHeader from './AuthenticatedHeader';
 import UnauthenticatedHeader from './UnauthenticatedHeader';
 
+const initialState = {
+  auth: {
+    user: {},
+    isAuthenticated: false,
+    loading: false,
+    error: {},
+  },
+};
+
+const mockStore = configureMockStore();
+const store = mockStore(initialState);
+
 describe('<AuthenticatedHeader /> Component', () => {
   it('should render the Authenticated Header without crashing', () => {
-    const wrapper = shallow(<AuthenticatedHeader />);
-    expect(wrapper.find('div').length).toEqual(1);
+    const wrapper = mount(<Provider store={store}><Router><AuthenticatedHeader /></Router></Provider>);
```

Comment:
Why???

Reply:
My test for the component requires I mount it which requires a store context. Hence why I used the provider to make the store context available. Without a provider, the test wouldn't pass

Reply:
Did you try to pass the store as a prop to the test component?

Reply:
yes I did. but I kept getting this error: `Either wrap the root component in a <Provider>, or pass a custom React context provider to <Provider> and the corresponding React context consumer to Connect(Header) in connect options.`

But it seems I know what I did wrong, I'll remove the provider and try again.

Reply:
MODIFIED: I have updated it, and everything is passing